zellij-server/src/panes/tiled_panes/mod.rs

```diff
@@ -917,7 +918,35 @@ impl TiledPanes {
                     *self.display_area.borrow(),
                     *self.viewport.borrow(),
                 );
-                let next_index = pane_grid.next_selectable_pane_id_to_the_left(&active_pane_id);
+                let mut next_index = pane_grid.next_selectable_pane_id_to_the_left(&active_pane_id);
+
+                if next_index.is_none() && wrap_around {
+                    let max_x = self
+                        .panes
+                        .iter()
+                        .filter(|(_, pane)| pane.selectable())
+                        .max_by_key(|(_, pane)| pane.x())
+                        .map(|(_, pane)| pane.x());
+
+                    next_index = if let Some(max_x) = max_x {
+                        let candidates = self
+                            .panes
+                            .values()
+                            .filter(|pane| pane.x() == max_x && pane.selectable())
+                            .collect::<Vec<_>>();
+
+                        let rightmost = candidates.iter().min_by_key(|pane| {
+                            (pane.y() as isize - self.panes[&active_pane_id].y() as isize).abs()
+                        });
+                        Some(
+                            rightmost
+                                .expect("Rightmost candidate should not be empty")
```

Comment:
Any chance we can somehow merge all of these options to a `map` chain?
Something like (pseudocode): `next_to_the_right().or_else(|| leftmost()).map(|| /* more logic here */)`?

Reply:
I made some changes to merge all into one big iterator call